lang/java/reef-applications/reef-vortex/src/main/java/org/apache/reef/vortex/driver/DefaultVortexMaster.java

```diff
@@ -140,4 +153,30 @@ public void taskletCancelled(final String workerId, final int taskletId) {
   public void terminate() {
     runningWorkers.terminate();
   }
+
+  private synchronized void putDelegate(final Tasklet tasklet, final VortexFutureDelegate delegate) {
+    taskletFutureMap.put(tasklet.getId(), delegate);
+  }
+
+  private synchronized VortexFutureDelegate fetchDelegate(final List<Integer> taskletIds) {
+    synchronized (taskletFutureMap) {
+      VortexFutureDelegate delegate = null;
+      for (final int taskletId : taskletIds) {
+        final VortexFutureDelegate currDelegate = taskletFutureMap.remove(taskletId);
+        if (currDelegate == null) {
+          // TODO[JIRA REEF-500]: Consider duplicate tasklets.
+          throw new RuntimeException("Tasklet should only be removed once.");
+        }
+
+        if (delegate == null) {
+          delegate = currDelegate;
+        } else {
+          assert delegate == currDelegate;
```

Comment:
It seems that multiple Tasklets should be associated to the same `VortexFutureDelegate` which would be done in `enqueueTasklet`. But thinking of local aggregation, could we determine which Tasklets will run in one worker, so that we can associate them in one `VortexFutureDelegate` at that time? Or do you consider adding a mechanism for the local aggregation?